### DISCUSSION TITLE
Fine tuning not working on Spanish Tacotron2

### DISCUSSION THREAD

**Discussion by AngelGuevara7** [coqui-ai/TTS#2171] [General Q&A]
Hello! I'm new with coquiTTS.
I am trying to fine tune the spanish Tacotron2,  but no results so far.  The model is unable to change the original voice (female) to the new voice(male).
While training, some of the losses seem to increase instead of decrease (stopnet, avg_loss). I followed the steps in the documentation, but don't know what I'm doing wrong.

The dataset I'm using is composed of 127 audio samples from 1 to 10 seconds aprox. I have downsampled the audios from 44100Hz to 16000Hz to match with the pretrained spanish tacotron2 that I found in the repository. 

I tried to emulate the configuration from the pretrained model. I attach the config file and the trainer log. 
 [config.txt](https://github.com/coqui-ai/TTS/files/10093479/config.txt)
[trainer_0_log.txt](https://github.com/coqui-ai/TTS/files/10093541/trainer_0_log.txt)

I also attach some figures regarding de loss and alignment during training and evaluation.

Train:
![Train7](https://user-images.githubusercontent.com/93993439/204008627-d1cacf02-73eb-43f3-bf4d-a4b665f4e3e5.PNG)
![Train8](https://user-images.githubusercontent.com/93993439/204008628-a6dddac7-5b66-41d8-9775-17ea8c8a6721.PNG)
![Train1](https://user-images.githubusercontent.com/93993439/204008631-9a0bbba4-ef76-46b8-a4f1-b97b9cce3c48.PNG)
![Train2](https://user-images.githubusercontent.com/93993439/204008634-becd13f9-1e8d-4dee-9eef-3e31ee1dc79c.PNG)
![Train3](https://user-images.githubusercontent.com/93993439/204008635-7d41c150-db3d-4908-bcbd-0ed52edef1a2.PNG)
![Train4](https://user-images.githubusercontent.com/93993439/204008637-dc81986c-6a6a-4318-ae52-6755c2d1ff0f.PNG)
![Train5](https://user-images.githubusercontent.com/93993439/204008640-541757c7-5892-4f6d-9d46-99d9310bd596.PNG)
![Train6](https://user-images.githubusercontent.com/93993439/204008641-390bb7ae-8985-427f-b76b-54064b859944.PNG)

Eval: 
![Eval1](https://user-images.githubusercontent.com/93993439/204008022-d8312d9f-ddef-4a99-8ef1-8df8aa2c8ed9.PNG)
![Eval5](https://user-images.githubusercontent.com/93993439/204008050-84ebb3ad-78a3-4f42-b920-9b5de3ab8696.PNG)
![Eval6](https://user-images.githubusercontent.com/93993439/204008061-6890e6db-328b-497b-a34b-e2d65f38c228.PNG)
![Eval7](https://user-images.githubusercontent.com/93993439/204008068-a8486a74-d895-48ab-b938-edd4289a3dde.PNG)
![Eval3](https://user-images.githubusercontent.com/93993439/204008071-b3a3c820-aab3-4992-9cab-5640f3b9b543.PNG)
![Eval2](https://user-images.githubusercontent.com/93993439/204008075-0feb6a9c-9f4a-464a-8abd-56985cf33700.PNG)
![Eval4](https://user-images.githubusercontent.com/93993439/204008076-b79c4097-5f85-49cf-9956-872dcf93efe0.PNG)

I would appreciate some help. 
Thanks in advance!!

**Comment by WeberJulian** [comment]
Hum it looks to me that you just didn't train enough for now, some of the losses decoder and post-net are still going down. 2k step isn't a lot (compared to the 380k steps the model was trained). But yeah voice identity should have changed by now, can you upload some audio samples with the associated step number please?

**Reply by AngelGuevara7** [reply]
Thanks for the reponse!
Sure. I attach some audios samples at step 382257.

Evaluation: 

https://user-images.githubusercontent.com/93993439/204257023-24cb7918-f817-45aa-b11c-cc5ec1e3b591.mp4


Training: 

https://user-images.githubusercontent.com/93993439/204257045-40175bc0-89ba-454f-84e8-21feb95db2fe.mp4


Test: 

https://user-images.githubusercontent.com/93993439/204257062-240e5420-650b-41fd-b2b3-5955c54b3a69.mp4



There is something curious that I don't completely understand. Evaluation and Training samples begin to resemble to the target voice, but when I do inference the output is more like the pre-trained voice (which resembles to the test audios that I have uploaded).
Thanks in advance!

**Reply by Ca-ressemble-a-du-fake** [reply]
Do you already have a checkpoint file after 2k steps to infer against ? By default checkpoints are written every 10k steps. I just mean that you may be still infering with the previous model (not yet fine tuned).

**Reply by AngelGuevara7** [reply]
Yes, I changed the default value to have earlier checkpoints. 

In particular, this is an audio sample infered with the checkpoint at step 382200. 



https://user-images.githubusercontent.com/93993439/204284811-7ad5993a-db56-4a7e-a5d6-52fd8e1ccf28.mp4


As it can be appreciated, the voice is still too close to the original one from the pre-trained model.

**Reply by Ca-ressemble-a-du-fake** [reply]
Then I would say that you  have to train a bit more. I have checkpoints every 10k steps so I don't know what happens in-between, so if I were in your shoes I would try again at step 390k or 400k.

**Reply by AngelGuevara7** [reply]
Ok. I had stopped at 2k steps because it seemed to me that it wasn't improving much, but I'm going to resume the training till 10k steps and check what happen then.  


I'll let you know if it makes any improvement.
Thanks for the help!
